packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -476,6 +476,10 @@ function isConditionalTest(node: TSESTree.Node): boolean {
     return isConditionalTest(parent);
   }
 
+  if (parent.type === AST_NODE_TYPES.UnaryExpression) {
```

Comment:
be sure to check that it's really a `!` operator! Or you will get bugs on code like this 🙂 
```ts
declare const a: string | undefined;
declare const b: string;
if (+(a || b)) {
}
```

Reply:
I made a mistake because I was not familiar with the type. I've also added a corresponding test case just in case.